### DISCUSSION TITLE
I wonder if DDT4ALL would work on a Nissan e-power Car?

### DISCUSSION THREAD

**Discussion by AndyIreland** [cedricp/ddt4all#1088] [General]
we might be getting a 2018 Nissan Note e-power petrol hybrid (Japanese import) - I wonder what the chances of DDT4ALL working on the ECU's ? and being 2018 I wonder if the ECU's will be Security Gateway covered or did that happen on later vehicles?

**Comment by KarelSvo** [comment]
medium to high

**Reply by AndyIreland** [reply]
medium to high chance that DDT4ALL will work on it or Medium to High chance that it will be blocked by security gateway? thanks

**Comment by KarelSvo** [comment]
There are some uncounted ddt ECU files for Nissan without the functions integrated into a screen. Pyren has the ability to display this well.
![img_530](https://github.com/cedricp/ddt4all/assets/30380693/42c9bd38-24b9-42e9-ba28-386797cd63d2)

![img_531](https://github.com/cedricp/ddt4all/assets/30380693/ee42398b-624a-4f3e-b872-1d8fe110f8be)

In ddt4all access via the manual request function.

![IMG_566](https://github.com/cedricp/ddt4all/assets/30380693/d1844e36-257f-4c88-ad5e-4548b1448c89)

**Reply by AndyIreland** [reply]
Thank you - yes I see that in DDT4ALL that the car was listed but 'screens' were blank , was wondering about PyREN as well.

**Comment by KarelSvo** [comment]
[ddt4all PY2](https://github.com/KarelSvo?tab=repositories)
edited vehicle list
current ecus, snat, dnat
File editor removed and renamed to Demo
XP-11
+Ecu names according to the CLIP database
+ddt2000 database old ecu versions removed
+dangerous commands in Radio control units removed

![IMG_567](https://github.com/cedricp/ddt4all/assets/30380693/afb3987a-98fc-4b48-a11d-4f4dfc41f08f)


You call your 3.0.1 release stable.
Connect ddt4all 3.0.1 to a USB interface that is powered via USB. Example OBDlink SX or vgate Vlinker FS.
Select xJA and then press the Scan button.
Select xJAPh2 then press the Scan button.
Show please photos of both results in the terminal.
I want to see if my experiences are the same.

**Reply by AndyIreland** [reply]
thanks I will give it a try out and let you know

**Reply by Furtif** [reply]
lol

**Reply by Furtif** [reply]
https://github.com/cedricp/ddt4all/discussions/1087#discussioncomment-9310404

**Reply by Furtif** [reply]
![image](https://github.com/cedricp/ddt4all/assets/11375035/5a83e393-0185-4521-9d36-0b92a5bf94e9)

0 ZERO x2b1 nissan note audio.....

**Reply by Furtif** [reply]
HE12 and others not in ddt2000 database [Nissan e-Power](https://customs.gov.mv/d/AR003-2019%20_e-Power%20Electric%20Car%20(NISSAN%20Note%20HE12).pdf)

**Reply by Furtif** [reply]
[Nissan introduces new electric-motor drivetrain: e-POWER](https://global.nissannews.com/en/releases/161102-02-e) leaf is in .. but...

**Reply by Furtif** [reply]
```
Initialisation ELM a la vitesse 115200...
Tentative d'ouverture du port COM4 @ 115200
Serial<id=0x150b00a4dc0, open=True>(port='COM4', baudrate=115200, bytesize=8, parity='N', stopbits=1, timeout=5, xonxoff=False, rtscts=False, dsrdtr=False)
Connexion OBDLink OK, tentative de commutation UART à pleine vitesse
OBDLINK a commuté le baurate OK, changeant la vitesse UART maintenant...
Connexion pleine vitesse OBDLink OK
Version OBDLink STN2230 v5.10.3```

**Comment by AndyIreland** [comment]
I hooked up the ol' ELM327 today (Nissan Note 2019 e-power hybrid) and fired up DDT4ALL and scanned can. 
This is what I got in bottom window . 
![image](https://github.com/cedricp/ddt4all/assets/49172217/1dacf83f-2592-409f-afef-1b42863ed78e)
but after 100% scanned no ECU's showed up in left window unfortunately :( - it remained blank.

here is text of what ecu's it found:

Using custom DDT database
Initilizing ELM with speed 38400...
Trying to open portCOM3 @ 38400
Checking port speed: 38400
Start COM speed : 38400
3649 loaded ECUs in database.
Scanning ECU DASHBOARD ADAPTATION BOX
Scanning ECU MSB
Scanning ECU INSULATOR
Scanning ECU FRONT RADAR/ACC
Scanning ECU CHECK CHASSIS/IDM
Scanning ECU UCH/BCM
Scanning ECU UPC/EMM
Scanning ECU 4 WHEELS STEERING
Scanning ECU AIRBAG STOP HV
Scanning ECU AIR CONDITIONING
Scanning ECU Central Display Module
Scanning ECU CTRL
Scanning ECU Emergency Service
Scanning ECU Cluster
Scanning ECU ASBMD
Scanning ECU SCCM
Scanning ECU SVS
Scanning ECU Touchpad, Telephone Key pad
Scanning ECU HeadUnit
Scanning ECU SCRCM
Scanning ECU RADIO
Scanning ECU ABS/ESC 29BIT
Scanning ECU UNCOUPLED BRAKE PEDAL
Scanning ECU INTELLIGENT KEY
Scanning ECU DRIVER'S SEAT
Scanning ECU LASER RADAR
Scanning ECU AIRBAG/PRETENSIONERS
Scanning ECU BSG
Scanning ECU Additional Heater
Scanning ECU MULTIMEDIA INTERFACE
Scanning ECU COMMUNICATIONS UNIT/NAV
Scanning ECU BSG 48v
Scanning ECU E-SHIFT
Scanning ECU INSTRUMENT PANEL
Scanning ECU SUNROOF CONTROL UNIT
Scanning ECU LEFT_HEAD_LAMP
Scanning ECU RIGHT_HEAD_LAMP
Scanning ECU ADAPTAVIVE DRIVING BEAM UNIT
Scanning ECU Audio
Scanning ECU CCGW
Scanning ECU 4 WHEELS DRIVE
Scanning ECU SWSP
Scanning ECU REV COUNTER
Scanning ECU TELEPHONE INDUCTION CHARGER
Scanning ECU NOISE CONTROL
Scanning ECU VEHICULE SOUND FOR PEDESTRIAN
Scanning ECU AMPLIFIER
Scanning ECU MULTIMEDIA CONNECTION
Scanning ECU PILOTED SUSPENSION
Scanning ECU ABS/ESC
Scanning ECU ENERGY CONTROL UNIT
Scanning ECU HIGH BEAM LIGHTING
Scanning ECU POWER ASSISTED STEERING
Scanning ECU TYRES/TPMS
Scanning ECU HYBRID ENGINE
Scanning ECU SCCM_29
Scanning ECU EPS
Scanning ECU INV-RE1_29b
Scanning ECU SU
Scanning ECU UREA DOSING CONTROL MODULE 29BIT
Scanning ECU SEALED FUEL TANK 29BIT
Scanning ECU TRACTION CONTROL UNIT 29BIT
Scanning ECU BRAKE BOOSTER 29BIT
Scanning ECU TACHOGRAPH
Scanning ECU DIFFERENTIAL LOCK
Scanning ECU TCASE
Scanning ECU HU
Scanning ECU BMS 12V
Scanning ECU HFCK
Scanning ECU PARKING BRAKE
Scanning ECU PARKING PROX CAN
Scanning ECU SUPERVISOR/DCDC
Scanning ECU HEAD UP DISPLAY
Scanning ECU AROUND VIEW MONITORING
Scanning ECU FRONT CAMERA
Scanning ECU LEFT REAR SIDE RADAR
Scanning ECU RIGHT REAR SIDE RADAR
Scanning ECU CHARGER
Scanning ECU HV CHARGER/BCB
Scanning ECU INVERTER/PEB
Scanning ECU DMC
Scanning ECU HANDS-FREE MODULE
Scanning ECU SOW Front Right
Scanning ECU SOW Front Left
Scanning ECU AIR CONDITIONING 2
Scanning ECU BMS/LBC
Scanning ECU BMS/LBC
Scanning ECU BMS Master
Scanning ECU EVC
Scanning ECU POWER LINE CONTROLLER
Scanning ECU DRIVING ASSISTANCE SUPERVISOR
Scanning ECU BLIND SPOT DETECTION
Scanning ECU BIRD EYE VIEW
Scanning ECU INJECTION 29BIT
Scanning ECU AUTOMATIC TRANSMISSION
Scanning ECU PASSENGER SEAT
Scanning ECU Magic Folding-RDCM
Scanning ECU AUTOMATIC TRANSMISSION 29BIT
Scanning ECU ELECTRIC TRACTION MOTOR 29BIT
Scanning ECU SUPERVISOR/DCDC 29BIT
Scanning ECU BCB/OBC 29BIT
Scanning ECU BMS/LBC 29BIT
Scanning ECU BMS SAFETY/LBC2 29BIT
Scanning ECU EVC 29BIT
Scanning ECU ADAPTATION UNIT
Scanning ECU ADAPTAVIVE DRIVING BEAM LEFT
Scanning ECU INJECTION
Scanning ECU ENGINEERING
Scanning ECU SCENE RECORDER
Scanning ECU LANE KEEP
Scanning ECU GATEWAY NISSAN
Scanning ECU STEERING
Scanning ECU DRIVER DOOR
Scanning ECU POWER BACK DOOR
Scanning ECU PASSENGER DOOR
Scanning ECU BMS SAFETY/LBC2
Scanning ECU VIRTUAL KEY MODULE
Scanning ECU TELEMATIC CONTROL UNIT
Scanning ECU ADAPTAVIVE DRIVING BEAM RIGHT
Scanning ECU GEARBOX ELEC CONTROL
Scanning ECU LPG/CNG
Scanning ECU DCM
Scanning ECU WIND SHIELD
Scanning ECU SCU_29b
Scanning ECU CHADeMo
Scanning ECU CENTRAL GATEWAY 4
Scanning ECU eCALL
Scanning ECU CENTRAL GATEWAY 3
Scanning ECU UREA DOSING CONTROL MODULE
Scanning ECU CENTRAL GATEWAY

Its a shame really, all I would like to try and configure DRL's to be on all the time, not just when sidelights are selected, and tail lights to come on with DTL, possibly welcome goodbye lights  and disable beeps when locking up and a few more things but if the ECU's dont show up I guess theres nothing I can do. 
If it is Gateway protected do you think this is why the ECU's are not showing up in the left window?
or would it be the case  that they would show up, but that you just wouldnt be able to change anything?
